app/assets/javascripts/components/ecnumbers.js

```diff
@@ -0,0 +1,260 @@
+import PriorityQueue from "../utilities/PriorityQueue.js";
+import {postJSON} from "../utils.js";
+/**
+ * @typedef {Object} FACounts
+ * @property {number} value count
+ * @property {string} name  The name of the GO/EC number
+ * @property {string} code  The code of the GO/EC number
+ */
+
+// const for private methods
+const addMissing = Symbol("[addMissing]");
+const addNames = Symbol("[addNames]");
+const addMissingNames = Symbol("[addMissingNames]");
+const artificial = Symbol("artificial");
+let ecNames = new Map([["-.-.-.-", "Enzyme Commission Numbers"]]);
+
+/**
+ * Class to contain the results of a fetch of EC numbers.
+ *
+ * - fetch information about missing numbers in hierarchy
+ * - Create a treeview
+ *
+ * The names associated with EC numbers are stored statically to prevent
+ * useless queries
+ */
+export default class ECNumbers {
+    /**
+     * Creates a summary of given EC numbers and their counts
+     * @param  {[FACounts]} ec list of EC numbers with their counts
+     */
+    constructor({numAnnotatedPeptides = null, data}) {
+        this.numTotalSet = numAnnotatedPeptides;
+        this.data = Array.from(data).sort((a, b) => (b.value - a.value));
+        this.ec = new this[addMissing](data);
+        ECNumbers[addNames](data);
+
+        // Fetch names in the background, not needed yet
+        setTimeout(()=>{
```

Comment:
spaces around `=>`, strange there is no eslint rule for this.

_[Original comment](https://github.ugent.be/unipept/unipept/pull/625#discussion_r13718) by @bmesuere on Mon Feb 19 2018 at 13:59._